db/db_iter.cc

```diff
@@ -159,7 +160,12 @@ class DBIter final: public Iterator {
   virtual bool Valid() const override { return valid_; }
   virtual Slice key() const override {
     assert(valid_);
-    return saved_key_.GetUserKey();
+    if(start_seqnum_ > 0) {
+      return saved_key_.GetInternalKey();
```

Comment:
how do you plan for user to deserialize internal key? Traditionally we've resisted exposing the serialization format in public headers, although I don't see any other path forward now.

Reply:
In my tests i'm using the following:

for (db_iter->SeekToFirst(); db_iter->Valid(); db_iter->Next()) {
      ParsedInternalKey ikey;
      ParseInternalKey(db_iter->key(), &ikey);
      // ikey.user_key.ToString()), ikey.type,  ikey.sequence;
    }

But that's inside RocksDB; I suppose client would need to have dbformat.h imported right?

Reply:
(on the other hand, I think it doesn't change db_iter public api since it returns a Slice, basically pointer to a chunk in the arena with metadata?)